### PR TITLE
[11.x] Defer registering schedule registered using  `ApplicationBuilder::withScheduling()`

### LINKED ISSUE #51354: Schedule gets registered with every artisan call
### Laravel Version

11.7.0

### PHP Version

8.3.6

### Database Driver & Version

_No response_

### Description

When running any `artisan` command (just `php artisan` does the same), the code inside `console.php` route file and/or `->withSchedule()` bootstrap method gets executed. Usually this wouldn't be a big deal, because the code should only register cron events. However, sometimes dynamic scheduling might involve database queries which can inflict unwanted performance penalty or in worst case scenario fail in CI setup.

We noticed this in our CI while upgrading from Laravel 10 to Laravel 11 (with slimmed down app structure). Our scheduler needs to be dynamic and it makes a DB query for that reason. When CI is setting up the application it runs `composer install` which then runs `php artisan package:discover --ansi` as a composer `post-autoload-dump` script. This is standard in every Laravel project. Since `package:discover` is an artisan command it boots up the scheduler as well. At this stage in CI database is not yet available thus we get a `QueryException: SQLSTATE[HY000] [2002] Connection refused`.

I have checked and this wasn't the case with Laravel 10. I'm not sure if this is now intended or not, but I would expect scheduler code to only execute when calling `php artisan schedule:run` and `php artisan schedule:work`.

### Steps To Reproduce

* Set up new Laravel 11 project.
* Add `dd();` inside `console.php` route file or inside `->withSchedule()` app bootstrap method.
* Run `php artisan` and observe the command to fail.

### PR DESCRIPTION
**Pull request by crynobone**
fixes https://github.com/laravel/framework/issues/51354

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -106,7 +106,7 @@
         "league/flysystem-sftp-v3": "^3.0",
         "mockery/mockery": "^1.6",
         "nyholm/psr7": "^1.2",
-        "orchestra/testbench-core": "^9.0.15",
+        "orchestra/testbench-core": "9.x-dev",
```
Comment (crynobone) [inline composer.json:109]:
```suggestion
        "orchestra/testbench-core": "^9.1.3",
```

File: composer.json [hunk 2]
```diff
@@ -106,7 +106,7 @@
         "league/flysystem-sftp-v3": "^3.0",
         "mockery/mockery": "^1.6",
         "nyholm/psr7": "^1.2",
-        "orchestra/testbench-core": "^9.0.15",
+        "orchestra/testbench-core": "^9.1.3",
```
Comment (crynobone) [inline composer.json:109]: This fixed discovering commands via Testbench's Console Kernel (used in tests below)

**Comment by taylorotwell** [comment]
I worry this will just break other applications. 😕

**Comment by EgorGruzdev** [comment]
This fix would help to avoid problems in the case of scheduling, not only through the settings in the code, but also, for example, through the database, at the moment, if the database is not available, then it is impossible to run atrisan, because running the schedule, for example, in our case, causes a connection to the database and redis, which may be unavailable and an error occurs.

And this edit allows you to avoid this.

**Pull request closed without merging**